Source/WebCore/platform/network/curl/NetworkStorageSessionCurl.cpp

```diff
@@ -135,10 +135,9 @@ HTTPCookieAcceptPolicy NetworkStorageSession::cookieAcceptPolicy() const
     return HTTPCookieAcceptPolicy::OnlyFromMainDocumentDomain;
 }
 
-std::pair<String, bool> NetworkStorageSession::cookiesForDOM(const URL& firstParty, const SameSiteInfo&, const URL& url, std::optional<FrameIdentifier>, std::optional<PageIdentifier>, IncludeSecureCookies, ApplyTrackingPrevention, ShouldRelaxThirdPartyCookieBlocking) const
+std::pair<String, bool> NetworkStorageSession::cookiesForDOM(const URL& firstParty, const SameSiteInfo&, const URL& url, std::optional<FrameIdentifier>, std::optional<PageIdentifier>, IncludeSecureCookies includeSecureCookies, ApplyTrackingPrevention, ShouldRelaxThirdPartyCookieBlocking) const
 {
-    // FIXME: This should filter secure cookies out if the caller requests it.
-    return { cookiesForSession(*this, firstParty, url, false), false };
+    return { cookiesForSession(*this, firstParty, url, false, includeSecureCookies), false };
```

Comment:
The second argument is used for `document.setSecureCookiesAccessed`. It seems important. https://commits.webkit.org/192473@main It'd be nice, if you properly calculate the second argument in this time.

Reply:
We can follow-up on this one since its not related to the bug report and I didn't find an easy way of doing it.